test/components/views/rooms/RoomHeader-test.tsx

```diff
@@ -0,0 +1,252 @@
+        expect(image.src).toEqual("data:image/png;base64,00");
+    });
+
+    test('shows the room avatar in a room with >2 people', () => {
+        // When we render a non-DM room with 3 people in it
+        const room = createRoom(
+            { name: "Z Room", isDm: false, users: ["other1", "other2"] });
+        const rendered = render(room);
+
+        // Then the room's avatar is the initial of its name
+        const initial = findSpan(rendered, ".mx_BaseAvatar_initial");
+        expect(initial.innerHTML).toEqual("Z");
+
+        // And there is no image avatar (because it's not set on this room)
+        const image = findImg(rendered, ".mx_BaseAvatar_image");
+        expect(image.src).toEqual("data:image/png;base64,00");
+    });
+
+    test('shows the room avatar in a DM with only ourselves', () => {
+        // When we render a non-DM room with 1 person in it
+        const room = createRoom({ name: "Z Room", isDm: true, users: [] });
+        const rendered = render(room);
+
+        // Then the room's avatar is the initial of its name
+        const initial = findSpan(rendered, ".mx_BaseAvatar_initial");
+        expect(initial.innerHTML).toEqual("Z");
+
+        // And there is no image avatar (because it's not set on this room)
+        const image = findImg(rendered, ".mx_BaseAvatar_image");
+        expect(image.src).toEqual("data:image/png;base64,00");
+    });
+
+    test('shows the user avatar in a DM with 2 people', () => {
+        // Note: this is the interesting case - this is the ONLY
+        //       time we should use the user's avatar.
+
+        // When we render a DM room with only 2 people in it
+        const room = createRoom(
+            { name: "Y Room", isDm: true, users: ["other"] });
+        const rendered = render(room);
+
+        // Then we use the other user's avatar as our room's image avatar
+        const image = findImg(rendered, ".mx_BaseAvatar_image");
+        expect(image.src).toEqual(
+            "http://this.is.a.url/example.org/other");
+
+        // And there is no initial avatar
+        expect(
+            rendered.querySelectorAll(".mx_BaseAvatar_initial"),
+        ).toHaveLength(0);
+    });
+
+    test('shows the room avatar in a DM with >2 people', () => {
+        // When we render a DM room with 3 people in it
+        const room = createRoom({
+            name: "Z Room", isDm: true, users: ["other1", "other2"] });
+        const rendered = render(room);
+
+        // Then the room's avatar is the initial of its name
+        const initial = findSpan(rendered, ".mx_BaseAvatar_initial");
+        expect(initial.innerHTML).toEqual("Z");
+
+        // And there is no image avatar (because it's not set on this room)
+        const image = findImg(rendered, ".mx_BaseAvatar_image");
+        expect(image.src).toEqual("data:image/png;base64,00");
+    });
+});
+
+interface IRoomCreationInfo {
+    name: string;
+    isDm: boolean;
+    users: string[];
```

Comment:
```suggestion
    userIds: string[];
```

We have a `User` class, so I think we can be more clear be calling this `userIds`

Reply:
Yes, nice - fixed.